lib/Conversion/TorchToLinalg/Uncategorized.cpp

```diff
@@ -189,25 +189,61 @@ static Value buildUnitNormalCdf(OpBuilder &b, Location &loc, Value x) {
   return buildNormalCdf(b, loc, x, zero, one);
 }
 
+template<typename MathOpTy>
+static Value createCalculationForMathOpWithDtypeConversion(
+              OpBuilder &b, TypeConverter *converter,
+              Value payloadArg, Operation *op) {
+  Type dtype = converter->convertType(op->getResult(0).getType())
+                    .template cast<RankedTensorType>()
+                    .getElementType();
+  Location loc = op->getLoc();
+  Value arg = convertScalarToDtype(b, loc, payloadArg, dtype);
+  return b.create<MathOpTy>(loc, arg);
+}
+
 static Value createLinalgPayloadCalculationForElementwiseOp(
     OpBuilder &b, Location loc, TypeConverter *converter,
     ValueRange payloadArgs, Operation *op, ArrayRef<Value> operands) {
-  if (isa<AtenTanhOp>(op))
-    return b.create<math::TanhOp>(loc, payloadArgs[0]);
-  if (isa<AtenExpOp>(op))
-    return b.create<math::ExpOp>(loc, payloadArgs[0]);
   if (isa<AtenFloorOp>(op))
     return b.create<math::FloorOp>(loc, payloadArgs[0]);
   if (isa<AtenCeilOp>(op))
     return b.create<math::CeilOp>(loc, payloadArgs[0]);
-  if (isa<AtenLogOp>(op))
-    return b.create<math::LogOp>(loc, payloadArgs[0]);
-  if (isa<AtenErfOp>(op))
-    return b.create<math::ErfOp>(loc, payloadArgs[0]);
-  if (isa<AtenSqrtOp>(op))
-    return b.create<math::SqrtOp>(loc, payloadArgs[0]);
-  if (isa<AtenRsqrtOp>(op))
-    return b.create<math::RsqrtOp>(loc, payloadArgs[0]);
+  if (isa<AtenTanhOp>(op)) {
+    return createCalculationForMathOpWithDtypeConversion<math::TanhOp>
+                                  (b, converter, payloadArgs[0], op);
```

Comment:
Also, I don't think `git clang-format HEAD~1` formatted successfully. This indentation does not look right to me. (note, you do have to re-commit the modified changes made by `clang-format)

Reply:
Oh, sorry. I forgot. Will do that!